cpp/src/arrow/array.h

```diff
@@ -34,10 +34,15 @@
 /// @{
 /// @}
 
+/// \defgroup encoded-arrays Concrete classes for encoded arrays
+/// @{
+/// @}
+
 #include "arrow/array/array_base.h"       // IWYU pragma: keep
 #include "arrow/array/array_binary.h"     // IWYU pragma: keep
 #include "arrow/array/array_decimal.h"    // IWYU pragma: keep
 #include "arrow/array/array_dict.h"       // IWYU pragma: keep
+#include "arrow/array/array_encoded.h"    // IWYU pragma: keep
```

Comment:
Maybe `array_rle` or `array_ree` would be a better name for RunEndEncoded arrays

DictionaryArrays are also a form of encoding so calling this `array_encoded` seems a bit vague